robot/impl/local_robot.go

```diff
@@ -900,6 +900,22 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 	var allErrs error
 
 	newConfig = r.updateDefaultServiceNames(newConfig)
+
+	// Sync Packages before reconfiguring rest of robot and resolving references to any packages
+	// in the config.
+	// TODO(RSDK-1849): Make this non-blocking so other resources that do not require packages can run before package sync finishes.
+	err := r.packageManager.Sync(ctx, newConfig.Packages)
+	if err != nil {
+		allErrs = multierr.Combine(allErrs, err)
+	}
+
+	err = r.replacePackageReferencesWithPaths(newConfig)
```

Comment:
Add tests for both.

In `local_robot_test.go`, I just tested that when we have package references in the config, we can still start the robot. As we discussed, I didn't test the integration with the transform camera, since this would require us to have actual packages in the tests and a more sophisticated implementation of the package manager for testing. I did however do extensive testing of this manually.